src/battle_util.c

```diff
@@ -11167,6 +11254,8 @@ u32 CalcSecondaryEffectChance(u8 battler, u8 secondaryEffectChance)
 {
     if (GetBattlerAbility(battler) == ABILITY_SERENE_GRACE)
         secondaryEffectChance *= 2;
+    if (gSideTimers[GetBattlerSide(battler)].rainbowTimer && gCurrentMove != MOVE_SECRET_POWER)
+        secondaryEffectChance *= 2;
```

Comment:
Two problems with this.
1. I think this is missing from the CFRU, but apparently, according to Bulbapedia, the rainbow can't double flinch chances. You'll need to modify this to take in a move and not just the secondaryEffectChance. 
2. You can't call `gCurrentMove` in this util function. It's not guaranteed it will be called at attack time. For example, the AI could use it in its calls. Instead, change the parameter to a move (like in the first suggestion) and check that instead.

Reply:
> the rainbow can't double flinch chances
Isn't it only in cases with Serene Grace?

Reply:
You're right, I misread. Seems CFRU is doing it correctly then. No change is needed then other than making a specific case for flinching.

Reply:
I changed the parameter to a move but would it be fine to keep `secondaryEffectChance` and keep checking the current move if a different function was made to account for the AI calls?
e.g.
```
CanBeParalyzed
AI_CanBeParalyzed
```

Reply:
applied